src/browser.ts

```diff
@@ -1,97 +1,57 @@
 import { JupyterFrontEnd } from '@jupyterlab/application';
 import { MainAreaWidget } from '@jupyterlab/apputils';
 import { PathExt } from '@jupyterlab/coreutils';
-import { DirListing, FileBrowser } from '@jupyterlab/filebrowser';
-import { Contents } from '@jupyterlab/services';
+import { IDocumentManager } from '@jupyterlab/docmanager';
+import { DocumentRegistry, IDocumentWidget } from '@jupyterlab/docregistry';
+import { FileBrowser } from '@jupyterlab/filebrowser';
+import { Kernel } from '@jupyterlab/services';
 
 import HDF5_FILE_TYPE from './fileType';
 import { h5webIcon } from './icons';
 import H5webWidgetFactory, { H5webWidget } from './widget';
 
+interface IPatchedDocumentManager extends IDocumentManager {
+  _classicOpen?: (
+    path: string,
+    widgetName?: string,
+    kernel?: Partial<Kernel.IModel>,
+    options?: DocumentRegistry.IOpenOptions
+  ) => IDocumentWidget | undefined;
+}
+
 const OPEN_H5WEB_COMMAND = 'h5web-open';
 
 // Patch the opening of HDF5 files. This is to bypass the core JupyterLab circuitry that fetches the base64 blob (see the comment in fileType.ts)
 const PATCH_OPENING = true;
 
-export function getBrowserListing(browser: FileBrowser): DirListing {
-  if ('listing' in browser) {
-    // @ts-ignore
-    return browser.listing; // JLab 3
-  }
-
-  // @ts-ignore
-  return browser._listing; // JLab 2
-}
-
-// Reimplementation of the double-click/Enter handling that adds the special handling for HDF files
-// https://github.com/jupyterlab/jupyterlab/blob/256d18253ec2733431a3289691b6a16766d4469c/packages/filebrowser/src/listing.ts#L1006
+// Add special handling for HDF files to the function handling file openings (docManager.open)
 export function patchOpeningOfHdf5File(
   app: JupyterFrontEnd,
-  browser: FileBrowser
+  docManager: IPatchedDocumentManager
 ): void {
   const { commands } = app;
-  const listing = getBrowserListing(browser);
 
-  // Reimplementation of handleOpen that add the special handling for HDF files
-  // https://github.com/jupyterlab/jupyterlab/blob/2.3.x/packages/filebrowser/src/listing.ts#L935
-  function handleOpen(item: Contents.IModel, event: Event): void {
-    if (!item) {
-      return;
-    }
-
-    const extname = PathExt.extname(item.path);
-    if (HDF5_FILE_TYPE.extensions.includes(extname)) {
+  docManager._classicOpen = docManager.open;
```

Comment:
To avoid patching the types of `docManager`, you may be able to do something like this:

```ts
const classicOpen = docManager.open;
classicOpen.call(docManager, path, ...otherParams);
```

Reply:
Thank you, I was looking for something along these lines